### PR TITLE
ROQA-118 add Signature Generation Jce Sampler

### PR DESCRIPTION
**Pull request by khanbu**
Signature Jce Sampler

### CODE REVIEW

**Comment by govardhanfortanix** [comment]
@khanbu Please run all the tests, we dont want to see any breaks.

File: README.md [hunk 1]
```diff
@@ -168,6 +201,13 @@ SDKMS Signature Generation | 1 | 300 | 16020 | 13 | 21 | 26 | 6 | 154 | 0.00% |
 
 * capacity columns is only supported for AES Encryption/Decryption with provided filepath.
 
+# Connection Pooling
+With JCE Connection Pooling
+
+The environment variable FORTANIX_CONN_MAX is set to the maximum number of connections from that instance of the JCE application.
+
+When the environment variable FORTANIX_CONN_MAX is not exported or is set to `0`, JCE will behave without any connection pooling/limit
```
Comment (khanbu) [inline README.md:209]: test conn pooling

File: src/main/java/com/fortanix/sdkms/performance/sampler/jce/SignatureVerificationJCESampler.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            sig.initSign(this.key.getPrivate());
+        } catch (InvalidKeyException e) {
+            LOGGER.log(Level.SEVERE, "failure in signature verification : " + e.getMessage(), e);
+            result.setSuccessful(false);
+        }
+        try {
+            byte[] data = this.input.getBytes("UTF8");
+            byte[] signatureBytes = sig.sign();
+            sig.initVerify(key.getPublic());
+            sig.update(ByteBuffer.wrap(data));
```
Comment (khanbu) [inline src/main/java/com/fortanix/sdkms/performance/sampler/jce/SignatureVerificationJCESampler.java:45]: Sign should be done just once, not for all threads

File: src/main/java/com/fortanix/sdkms/performance/sampler/jce/SignatureVerificationBaseSampler.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    public void setupTest(JavaSamplerContext context) {
+        super.setupTest(context);
+        String algorithm = context.getParameter(ALGORITHM, "RSA");
+        String defaultKeySize = algorithm.equalsIgnoreCase("EC") ? "NistP192" : "1024";
+        String keySize = context.getParameter(KEY_SIZE, defaultKeySize);
+        String filePath = context.getParameter(FILE_PATH);
+        String hashAlgorithm = context.getParameter(HASH_ALGORITHM, "SHA256");
+        this.pssParameterDigest = hashAlgorithm;
+        this.input = "random-text";
```
Comment (khanbu) [inline src/main/java/com/fortanix/sdkms/performance/sampler/jce/SignatureVerificationBaseSampler.java:36]: use min length 64 byte random text

**Comment by khanbu** [comment]
> @khanbu Please run all the tests, we dont want to see any breaks.

@govardhanfortanix I have tested by running all the commands mentioned in readme....Can you please elaborate what tests you are running to..which I should run for further testing

**Comment by govardhanfortanix** [comment]
@khanbu If all the tests that are in README works fine without any issues.  Then we are good.

**Comment by govardhanfortanix** [comment]
Changes look good !! we can merge this! 
@khanbu @vishwas-ftx @laxmishanagaraj

**Pull request merged**